sqlx-core/src/sqlite/types/uint.rs

```diff
@@ -0,0 +1,104 @@
+use std::convert::TryInto;
+
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::BoxDynError;
+use crate::sqlite::type_info::DataType;
+use crate::sqlite::{Sqlite, SqliteArgumentValue, SqliteTypeInfo, SqliteValueRef};
+use crate::types::Type;
+
+impl Type<Sqlite> for u8 {
+    fn type_info() -> SqliteTypeInfo {
+        SqliteTypeInfo(DataType::Int)
+    }
+
+    fn compatible(ty: &SqliteTypeInfo) -> bool {
+        matches!(ty.0, DataType::Int | DataType::Int64)
+    }
+}
+
+impl<'q> Encode<'q, Sqlite> for u8 {
+    fn encode_by_ref(&self, args: &mut Vec<SqliteArgumentValue<'q>>) -> IsNull {
+        args.push(SqliteArgumentValue::Int(*self as i32));
+
+        IsNull::No
+    }
+}
+
+impl<'r> Decode<'r, Sqlite> for u8 {
+    fn decode(value: SqliteValueRef<'r>) -> Result<Self, BoxDynError> {
+        Ok(value.int().try_into()?)
+    }
+}
+
+impl Type<Sqlite> for u16 {
+    fn type_info() -> SqliteTypeInfo {
+        SqliteTypeInfo(DataType::Int)
+    }
+
+    fn compatible(ty: &SqliteTypeInfo) -> bool {
+        matches!(ty.0, DataType::Int | DataType::Int64)
+    }
+}
+
+impl<'q> Encode<'q, Sqlite> for u16 {
+    fn encode_by_ref(&self, args: &mut Vec<SqliteArgumentValue<'q>>) -> IsNull {
+        args.push(SqliteArgumentValue::Int(*self as i32));
+
+        IsNull::No
+    }
+}
+
+impl<'r> Decode<'r, Sqlite> for u16 {
+    fn decode(value: SqliteValueRef<'r>) -> Result<Self, BoxDynError> {
+        Ok(value.int().try_into()?)
+    }
+}
+
+impl Type<Sqlite> for u32 {
+    fn type_info() -> SqliteTypeInfo {
+        SqliteTypeInfo(DataType::Int64)
+    }
+
+    fn compatible(ty: &SqliteTypeInfo) -> bool {
+        matches!(ty.0, DataType::Int | DataType::Int64)
+    }
+}
+
+impl<'q> Encode<'q, Sqlite> for u32 {
+    fn encode_by_ref(&self, args: &mut Vec<SqliteArgumentValue<'q>>) -> IsNull {
+        args.push(SqliteArgumentValue::Int64(*self as i64));
+
+        IsNull::No
+    }
+}
+
+impl<'r> Decode<'r, Sqlite> for u32 {
+    fn decode(value: SqliteValueRef<'r>) -> Result<Self, BoxDynError> {
+        Ok(value.int64().try_into()?)
+    }
+}
+
+impl Type<Sqlite> for u64 {
+    fn type_info() -> SqliteTypeInfo {
+        SqliteTypeInfo(DataType::Int64)
+    }
+
+    fn compatible(ty: &SqliteTypeInfo) -> bool {
+        matches!(ty.0, DataType::Int | DataType::Int64)
+    }
+}
+
+impl<'q> Encode<'q, Sqlite> for u64 {
+    fn encode_by_ref(&self, args: &mut Vec<SqliteArgumentValue<'q>>) -> IsNull {
+        args.push(SqliteArgumentValue::Int64(*self as i64));
```

Comment:
This will result in data loss for `u64` larger than `std::i64::MAX`.

Reply:
This should be `try_into()` but `Encode` is not fallible (yet).

Reply:
@dignifiedquire Does your use case _need_ `u64` ? It might be safer to remove this one until we have a fallible `Encode`.

Reply:
Ah yes, you are right. I am fine not having u64 supported for the moment, so likely better to remove this.